feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/presentation/staking/delegation/proxy/revoke/ConfirmRemoveStakingProxyViewModel.kt

```diff
@@ -0,0 +1,161 @@
+    private val assetFlow = selectedAssetState.selectedAssetFlow()
+        .flatMapLatest { assetUseCase.assetFlow(it) }
+        .shareInBackground()
+
+    val feeMixin: FeeLoaderMixin.Presentation = feeLoaderMixinFactory.create(assetFlow)
+
+    val chainModel = chainFlow.map { chain ->
+        mapChainToUi(chain)
+    }
+
+    val walletUiFlow = selectedMetaAccountFlow.map { walletUiUseCase.walletUiFor(it) }
+
+    val proxiedAccountModel = combine(selectedMetaAccountFlow, chainFlow) { metaAccount, chain ->
+        val address = metaAccount.requireAddressIn(chain)
+
+        generateAccountAddressModel(chain, address)
+    }
+
+    val proxyAccountModel = chainFlow.map { chain ->
+        generateAccountAddressModel(chain, payload.proxyAddress)
+    }
+
+    val validationProgressFlow = MutableStateFlow(false)
+
+    init {
+        loadFee()
+    }
+
+    fun backClicked() {
+        router.back()
+    }
+
+    fun proxiedAccountClicked() {
+        launch {
+            showExternalActions(proxiedAccountModel.first().address)
+        }
+    }
+
+    fun proxyAccountClicked() {
+        showExternalActions(payload.proxyAddress)
+    }
+
+    fun confirmClicked() = launch {
+        val metaAccount = accountRepository.getSelectedMetaAccount()
+        val chain = selectedAssetState.chain()
+        val validationPayload = RemoveStakingProxyValidationPayload(
+            chain = chain,
+            asset = assetFlow.first(),
+            proxyAddress = payload.proxyAddress,
+            proxiedAccountId = metaAccount.requireAccountIdIn(chain),
+            fee = feeMixin.awaitDecimalFee()
+        )
+
+        validationExecutor.requireValid(
+            validationSystem = removeStakingProxyValidationSystem,
+            payload = validationPayload,
+            validationFailureTransformer = { mapRemoveStakingProxyValidationFailureToUi(resourceManager, it) },
+            progressConsumer = validationProgressFlow.progressConsumer()
+        ) {
+            sendTransaction(it.chain, it.proxiedAccountId, it.chain.accountIdOf(it.proxyAddress))
+        }
+    }
+
+    private fun loadFee() {
+        launch {
+            val metaAccount = selectedMetaAccountFlow.first()
+            val chain = selectedAssetState.chain()
+            val proxiedAccountId = metaAccount.requireAccountIdIn(chain)
+
+            feeMixin.loadFee(
+                this,
+                chain.id,
+                feeConstructor = { removeStakingProxyInteractor.estimateFee(chain, proxiedAccountId) },
+                onRetryCancelled = ::backClicked
+            )
+        }
+    }
+
+    private fun sendTransaction(chain: Chain, proxiedAccount: AccountId, proxyAccount: AccountId) = launch {
+        val result = removeStakingProxyInteractor.removeProxy(chain, proxiedAccount, proxyAccount)
+
+        validationProgressFlow.value = false
+
+        if (result.isSuccess) {
+            router.returnToStakingMain()
+        }
```

Comment:
We should show something in case of failure too:

```kotlin
result.onSuccess {
           router.returnToStakingMain()
        }
            .onFailure(::showError)
```